java/runtime/src/main/java/org/ray/runtime/util/BinaryFileUtil.java

```diff
@@ -22,27 +24,42 @@
   public static final String CORE_WORKER_JAVA_LIBRARY =
       System.mapLibraryName("core_worker_library_java");
 
-  public static File getFile(String destDir, String fileName) {
-    File file = new File(String.format("%s/%s", destDir, fileName));
-    if (file.exists()) {
-      return file;
-    }
-
-    final File dir = file.getParentFile();
-    try {
-      if (!dir.exists()) {
+  /**
+   * Extract a resource file to <code>destDir</code>.
+   * Note that this a process-safe operation. If multi processes extract the file to same
+   * directory concurrently, this operation will be protected by a file lock.
+   *
+   * @param destDir  a directory to extract resource file to
+   * @param fileName resource file name
+   * @return extracted resource file
+   */
+  public static File getFileSafe(String destDir, String fileName) {
```

Comment:
how about keeping the name `getFile`? I think it's fine as long as "safe" is documented in the comment.

Reply:
fixed